x-pack/plugins/actions/server/lib/mustache_renderer.test.ts

```diff
@@ -340,4 +340,46 @@ describe('mustache_renderer', () => {
     const expected = '1 - {"c":2,"d":[3,4]} -- 5,{"f":6,"g":7}';
     expect(renderMustacheString('{{a}} - {{b}} -- {{e}}', deepVariables, 'none')).toEqual(expected);
   });
+
+  describe('converting dot variables', () => {
```

Comment:
What happens with `{a: 1, 'a.b': 2}`?  Presumably `a` is either `1` or `{b: 2}`.  Seems like `1` would be correct - "auto-objectifying" shouldn't destroy existing variables.

Reply:
Ah, but my test still fails.  

    expect(renderMustacheString('{{a}}', { a: 1, 'a.b': 2 }, 'none')).toEqual('1');

    ...
    Expected: "1"
    Received: "{\"b\":2}"

Thinking it's because that `1` isn't an object, so would also be a problem with strings and booleans.  A test with an object `{c: 3}` instead of `1` came back with `{c: 3, b: 2}`, as expected.

I wonder if this would ever really be a problem?  I'm thinking about ES multi-fields, like `name` and `name.keyword` coming back somewhere, as property names, both with string values.  I guess in this case they'd both be the same :-), but should `name` be `the-string` or should `name` be `{keyword: 'the-string'}`?  Feels like the "original" should win, but what if there's gold in those child objects!?!?!?

Reply:
Oh you are right! I am also wondering if that will be a problem. Do you think I should go ahead and try to fix this?

Reply:
TBH, not sure.  If this scenario did happen, we'd be exchanging a tree of objects for a single value - and that value is going to be a string/boolean/number.  So we'd be potentially "dropping" a lot of data.

If we went with the object version, but later had to make the single value available (somehow) as well, you could imagine adding something like `propName.__original__` to the variables, to store the original one.  If we went with the single value version and wanted to make the objects (some) version later, seems harder.

Any other approaches to this problem that might avoid this?  I happened to think of this "auto-expansion" as a neat trick that hopefully "works like you would expect", but perhaps there's another way ...  possibly some ideas in https://github.com/elastic/kibana/issues/84217 ...

I think I'm tending towards going with the current implementation, as is - with the object winning over the single value.  Let's add a simple test like the one I did, so we at least have a contract that it works that way :-)

Reply:
Okay that sounds good! I agree that it would be easier to add that in the future if we end up needing it. I will add that test, thank you!

Reply:
I added the new test here 1b282605f1fc864e47084325e0e50549f6304e0b